lib/govuk_app_config/default_configuration.rb

```diff
@@ -0,0 +1,86 @@
+require "govuk_app_config/govuk_data_sync"
+
+class DefaultConfiguration < SimpleDelegator
```

Comment:
This class should be under the GovukError namespace since there are few things going on in this gem.

I feel like this class has a little too much going on in it with it handling both the default configuration and our logic changes. I'd suggest that you instead have this class as entirely the delegator - So potentially GovukError::Configuration and then have a different file to do the configuration (could re-instate govuk_app_config/configure.rb though I feel like that isn't specific enough that it's govuk_error configuration, so could have a govuk_app_config/govuk_error/configure.rb)

I think that idea would also help in your testing as it seems a pain to test every part of the config (and somewhat superfluous) but you do want to test the logic you're adding. You might even find the data sync code can fit into a shorter version of this class.